Makefile

```diff
@@ -1,27 +1,57 @@
-CPP_API ?= .
+ifdef NO_CONTEXT
+  CPP_CONTEXT_LIB =
+else
+  CPP_CONTEXT_LIB = ${PROXY_WASM_CPP_SDK}/proxy_wasm_intrinsics.cc
+endif
 
-all: proxy_wasm_intrinsics.pb.h proxy_wasm_intrinsics_lite.pb.h struct_lite.pb.h ${CPP_API}/libprotobuf.a ${CPP_API}/libprotobuf-lite.a
+PROTOBUF?=none
+ifeq ($(PROTOBUF), full)
+  PROTO_DEPS := protobuf
+  PROTO_OPTS := -DPROXY_WASM_PROTOBUF_FULL=1 \
+		${PROXY_WASM_CPP_SDK}/proxy_wasm_intrinsics.pb.cc
+else ifeq ($(PROTOBUF), lite)
+  PROTO_DEPS := protobuf-lite
+  PROTO_OPTS := -DPROXY_WASM_PROTOBUF_LITE=1 \
+		${PROXY_WASM_CPP_SDK}/proxy_wasm_intrinsics_lite.pb.cc \
+		${PROXY_WASM_CPP_SDK}/struct_lite.pb.cc
+else
+  PROTO_DEPS :=
+  PROTO_OPTS :=
+endif
 
-protobuf: proxy_wasm_intrinsics.pb.h proxy_wasm_intrinsics_lite.pb.h struct_lite.pb.h
+# Provide a list of libraries that the wasm depends on (absl_*, re2, etc).
+WASM_DEPS?=absl_base
 
-proxy_wasm_intrinsics.pb.h: proxy_wasm_intrinsics.proto
-	protoc --cpp_out=. proxy_wasm_intrinsics.proto
+# Determine dependency link options.
+# NOTE: Strip out -pthread which RE2 claims to need...
+PKG_CONFIG?=pkg-config
+PKG_CONFIG_PATH=${EMSDK}/upstream/emscripten/cache/sysroot/lib/pkgconfig
+WASM_LIBS=$(shell $(PKG_CONFIG) $(WASM_DEPS) $(PROTO_DEPS) \
```

Comment:
Yep, it's not a perfect hack. Dupe targets doesn't hurt, but you're right that it might mangle libs ending in pthread. Best I can do for now (I tried a lot of ways to fix the RE2 pthread reference, no success).